### PR TITLE
Kotlin ktor framework

### PR DESCRIPTION
**Pull request by cy6erGn0m**
Add [ktor framework](https://github.com/Kotlin/ktor) written in Kotlin, supports multiple hosts to run on: netty, jetty, servlets, possibly more. For now only test with netty.

### CODE REVIEW

**Comment by nbrady-techempower** [comment]
Hi @cy6erGn0m Thanks for the submission!
Be sure to add a line in the `.travis.yml` for testing and please add a `README.md` similar to [this one](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Perl/mojolicious/README.md) so we can get some more information about the framework.

File: frameworks/Kotlin/ktor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.mchange</groupId>
+            <artifactId>c3p0</artifactId>
+            <version>0.9.5.2</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>6.0.3</version>
```
Comment (martin-g) [inline frameworks/Kotlin/ktor/pom.xml]: Version 6.x breaks at the official run servers.
Several frameworks have been downgraded to 5.x recently.

File: frameworks/Kotlin/ktor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            <version>${ktor.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.7</version>
+        </dependency>
+        <dependency>
+            <groupId>com.mchange</groupId>
+            <artifactId>c3p0</artifactId>
```
Comment (martin-g) [inline frameworks/Kotlin/ktor/pom.xml]: HikariCP is better.

File: frameworks/Kotlin/ktor/src/main/kotlin/org/jetbrains/ktor/benchmarks/Hello.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                            null -> result.single()
+                            else -> result
+                        }), ContentType.Application.Json, HttpStatusCode.OK))
+                    }
+                }
+            }
+            get("/updates") {
+                pool.connection.use { connection ->
+                    val queries = call.queries()
+                    val random = Random()
```
Comment (martin-g) [inline frameworks/Kotlin/ktor/src/main/kotlin/org/jetbrains/ktor/benchmarks/Hello.kt]: You may also use ThreadLocalRandom

**Comment by cy6erGn0m** [comment]
@martin-g thanks for the tips, have applied them !

**Comment by cy6erGn0m** [comment]
@nbrady-techempower added to travis list (all seems to be green), added readme

**Comment by martin-g** [comment]
@cy6erGn0m Thank you for your work on Kotlin!

**Comment by nbrady-techempower** [comment]
@cy6erGn0m Everything looks good! Thanks for the contribution!

**Pull request merged**